distribution/openrct2.d.ts

```diff
@@ -307,46 +475,71 @@ declare global {
         "research" |
         "interest";
 
+    /** Arugments for the GameActionEvent hook callback. */
     interface GameActionEventArgs {
+        /** The number of the player who executed the action. TODO always 0 in single player? */
         readonly player: number;
+        /** The event type. */
         readonly type: string;
+        /** TODO True if the event only occurs in the client, false if needs to be synchronized to the network? */
         readonly isClientOnly: boolean;
+        /** TODO */
         readonly args: object;
+        /** The result of the action. */
         result: GameActionResult;
     }
 
+    /** The result of the game action. */
     interface GameActionResult {
+        /** TODO? */
```

Comment:
This is a categorisation of the error. We should add an enum for it. Types can be `NOT_OWNED`, `INVALID_PARAMETER`, ...